magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleTransitionExtensions.kt

```diff
@@ -0,0 +1,77 @@
+package com.wealthfront.magellan.lifecycle
+
+import android.content.Context
+import com.wealthfront.magellan.lifecycle.LifecycleState.Created
+import com.wealthfront.magellan.lifecycle.LifecycleState.Destroyed
+import com.wealthfront.magellan.lifecycle.LifecycleState.Resumed
+import com.wealthfront.magellan.lifecycle.LifecycleState.Shown
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.BACKWARDS
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.FORWARD
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.NO_MOVEMENT
+
+public fun <T> T.transitionToState(newState: LifecycleState) where T : LifecycleAware, T : LifecycleOwner {
+  transition(this.currentState, newState)
+}
+
+public fun LifecycleAware.transition(oldState: LifecycleState, newState: LifecycleState) {
+  listOf(this).transition(oldState, newState)
+}
+
+public fun Iterable<LifecycleAware>.transition(oldState: LifecycleState, newState: LifecycleState) {
+  var currentState = oldState
+  while (currentState.order != newState.order) {
+    currentState = when (currentState.getDirectionForMovement(newState)) {
+      FORWARD -> next(this, currentState, newState.context!!)
+      BACKWARDS -> previous(this, currentState, oldState.context!!)
+      NO_MOVEMENT -> throw IllegalStateException("Attempting to transition from $currentState to $newState")
+    }
+  }
+}
```

Comment:
I took these out of LifecycleStateMachine because that class wasn't actually giving us any value, and this way is much more intuitive IMO